thrift/thrift0.13/src/main/java/com/linecorp/armeria/server/thrift/ThriftCallServiceBuilder.java

```diff
@@ -0,0 +1,132 @@
+ */
+
+package com.linecorp.armeria.server.thrift;
+
+import static com.google.common.collect.ImmutableMap.toImmutableMap;
+import static java.util.Objects.requireNonNull;
+
+import java.util.Map;
+import java.util.concurrent.Executors;
+
+import com.google.common.collect.ImmutableListMultimap;
+
+import com.linecorp.armeria.common.annotation.UnstableApi;
+
+/**
+ * A fluent builder to build an instance of {@link ThriftCallService}.
+ *
+ * <h2>Example</h2>
+ * <pre>{@code
+ * ThriftCallService service = ThriftCallService
+ *                 .builder()
+ *                 .addService(defaultServiceImpl) // Adds a service
+ *                 .addService("foo", fooServiceImpl) // Adds a service with a key
+ *                 .addService("foobar", fooServiceImpl)  // Adds multiple services to the same key
+ *                 .addService("foobar", barServiceImpl)
+ *                 .addService("foobarOnce", fooServiceImpl, barServiceImpl) // Adds multiple services at once
+ *                  // Adds multiple services by list
+ *                 .addService("foobarList", ImmutableList.of(fooServiceImpl, barServiceImpl))
+ *                  // Adds multiple services by map
+ *                 .addServices(ImmutableMap.of("fooMap", fooServiceImpl, "barMap", barServiceImpl))
+ *                  // Adds multiple services by map
+ *                 .addServices(ImmutableMap.of("fooIterableMap",
+ *                                              ImmutableList.of(fooServiceImpl, barServiceImpl)))
+ *                 .build();
+ * }</pre>
+ *
+ * @see ThriftCallService
+ */
+@UnstableApi
+public final class ThriftCallServiceBuilder {
+    private final ImmutableListMultimap.Builder<String, Object> servicesBuilder =
+            ImmutableListMultimap.builder();
+
+    private boolean useBlockingTaskExecutor;
+
+    ThriftCallServiceBuilder() {}
+
+    /**
+     * Adds multiple services by map for {@link ThriftServiceEntry}.
+     */
+    public ThriftCallServiceBuilder addServices(Map<String, ?> services) {
+        requireNonNull(services, "services");
+        if (services.isEmpty()) {
+            throw new IllegalArgumentException("empty services");
+        }
+        services.forEach((k, v) -> {
+            if (v instanceof Iterable<?>) {
+                servicesBuilder.putAll(k, (Iterable<?>) v);
+            } else {
+                servicesBuilder.put(k, v);
+            }
+        });
+        return this;
+    }
+
+    /**
+     * Adds a service for {@link ThriftServiceEntry}.
+     */
+    public ThriftCallServiceBuilder addService(Object... service) {
+        requireNonNull(service, "service");
+        return addService("", service);
+    }
+
+    /**
+     * Adds a service with a key for {@link ThriftServiceEntry}.
+     */
+    public ThriftCallServiceBuilder addService(String key, Object... service) {
+        requireNonNull(key, "key");
+        requireNonNull(service, "service");
+        servicesBuilder.putAll(key, service);
```

Comment:
Should we also throw an `IllegalArgumentException` if `service.length == 0`?

Reply:
Done!